src/swarm-tui/Swarm/TUI/Controller/UpdateUI.hs

```diff
@@ -0,0 +1,250 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+-- |
+-- SPDX-License-Identifier: BSD-3-Clause
+--
+-- The main TUI update logic that is called from other controller parts.
+module Swarm.TUI.Controller.UpdateUI (
```

Comment:
I love that you have also split out the `Controller` module into submodules --- it was definitely one of the most in need of splitting up.

Reply:
It was a lot of work; I wish HLS had refactorings to move and rename stuff between files, like JetBrains IDEs. 😅

Reply:
Agreed!